File: src/services/payrollService.js
import api from './api'

export const payrollService = {
  getAllPayrolls: async () => {
    const response = await api.get('/payroll')
    return response.data
  },

  getPayrollsByEmployee: async (employeeId) => {
    const response = await api.get(`/payroll/employee/${employeeId}`)
    return response.data
  },

  getPayrollsByPeriod: async (month, year) => {
    const response = await api.get('/payroll/period', {
      params: { month, year }
    })
    return response.data
  },

  getPayrollById: async (id) => {
    const response = await api.get(`/payroll/${id}`)
    return response.data
  },

  createPayroll: async (payrollData) => {
    const response = await api.post('/payroll', payrollData)
    return response.data
  },

  generateAllPayrolls: async (month, year) => {
    const response = await api.post('/payroll/generate-all', null, {
      params: { month, year }
    })
    return response.data
  },

  processPayroll: async (id) => {
    const response = await api.patch(`/payroll/${id}/process`)
    return response.data
  },

  deletePayroll: async (id) => {
    await api.delete(`/payroll/${id}`)
  }
}

File: PayrollManagementSystem/src/main/java/com/payroll/service/PayrollService.java
// PayrollService.java
package com.payroll.service;

import com.payroll.dto.PayrollDto;
import java.util.List;

public interface PayrollService {
    List<PayrollDto> getAllPayrolls();
    List<PayrollDto> getPayrollsByEmployee(Long employeeId);
    List<PayrollDto> getPayrollsByMonthAndYear(Integer month, Integer year);
    PayrollDto getPayrollById(Long id);
    PayrollDto createPayroll(PayrollDto payrollDto);
    List<PayrollDto> generateAllPayrolls(Integer month, Integer year);
    PayrollDto processPayroll(Long id);
    void deletePayroll(Long id);
}
